server/pbench/bin/pbench-verify-s3-backup-tarballs

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import glob
+import hashlib
+import boto3
+import configtools
+
+options = []
+opts, args = configtools.parse_args(options)
+conf, files = configtools.init(opts)
+
+# configuration file
+endpoint_url = conf.get('aws-s3', 'endpoint_url')
+aws_access_key_id = conf.get('aws-s3', 'aws_access_key_id')
+aws_secret_access_key = conf.get('aws-s3', 'aws_secret_access_key')
+bucket_name = conf.get('aws-s3', 'bucket_name')
+target_dir = conf.get('aws-s3', 'target_dir')
+
+s3 = boto3.resource('s3', aws_access_key_id='%s' % aws_access_key_id,
+                    aws_secret_access_key='%s' % aws_secret_access_key, endpoint_url='%s' % endpoint_url)
+
+def checkmd5(target_dir, list_dir, indicator):
+    tarlist = glob.glob('%s/*/*.tar.xz' % target_dir)
+    for tar in tarlist:
+        controller = tar.split("/")[-2]
+        result_name = tar.split("/")[-1]
+        md5 = ("%s.md5" % tar)
+        with open("%s/list.%s" % (list_dir, indicator), 'a') as f:
+                f.write("%s\n" % result_name)
+        with open(md5) as f:
+            md5_value = f.readline().split(" ")[0]
+        with open(tar, 'rb') as f:
+            data = f.read()
+            md5_returned = hashlib.md5(data).hexdigest()
+        if md5_value == md5_returned:
+            with open("%s/ok_list.%s" % (list_dir, indicator), 'a') as f:
+                f.write("%s/%s\n" % (controller, result_name))
+        elif md5 != md5_returned:
+            with open("%s/fail_list.%s" % (list_dir, indicator), 'a') as f:
+                f.write("%s/%s\n" % (controller, result_name))
+
+
+def getlist_s3(bucket_name, list_dir):
+    for result in s3.Bucket(name=bucket_name).objects.all():
+        with open("%s/s3_list" % list_dir, 'a') as f:
```

Comment:
Are you sure you want to open and close the s3_list file each time through the loop?

Perhaps try:

```
with open(os.path.join(list_dir, "s3_list"), 'w') as fp:
    for result in s3.Bucket(name=bucket_name).objects.all():
        fp.write("%s\n" % result.key)
```

Reply:
Does the `result` entry also provide the `ETag` or MD5 value?  If so, let's add it to the list and check that the MD5 value in S3 matches our local copy.

If not (which I hope is not the case), we'll want to fetch that MD5 value for all tarballs in the S3 bucket and add it explicitly.

Reply:
Put more stuff in the function for checking md5. Done